kernel/kernel-impl/src/main/java/org/sakaiproject/authz/impl/SakaiSecurity.java

```diff
@@ -479,6 +504,20 @@ String makeCacheKey(String userId, String function, String reference, boolean is
                 }
             }
         }
+	if (checkSwapped) {
```

Comment:
Yes, it looks like I could remove the creation and test of auth. The rest is necessary to make sure I get the argument to pass to getUserEffectiveRole the same way that DbAuthzGroupService does. My concern is that the reference isn't necessarily always /site/NNN. DbAuthzGroupService will check any reference that has the realm ID anywhere in it and use the effective role. If I don't do the same check, I could call getUserEffectiveRole with an argument that won't match. E.g. suppose /content/group/NNNN or /realm/site/NNN is passed? I'd love to hear that this isn't necessary, but I couldn't prove that it wasn't.